CONTRIBUTING.md

```diff
@@ -27,3 +26,25 @@ Specifically:
 - We don't merge code into `main` that isn't releasable
 - We perform automated testing on all changes before they get merged to `main`
 - We create immutable release artifacts
+
+## How to Contribute
+Please ensure there is a Gitub issue for your proposed change, this helps the UDS CLI team to understand the context of the change and to track the progress of the work. If there isn't an issue for your change, please create one before starting work. The recommended workflow for contributing is as follows:
+
+1. **Fork this repo** and clone it locally
+2. **Create a branch** for your changes
+3. **Create and [test](#testing)** your changes
```

Comment:
Maybe not necessary in this PR but probably good to also add a note about documentation (i.e. documenting new features added in a PR)?

Reply:
added!